sdk/iot/Azure.Iot.Hub.Service/src/DevicesClient.cs

```diff
@@ -13,13 +17,489 @@ namespace Azure.Iot.Hub.Service
+        /// <param name="precondition">The condition for which each operation will execute.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The result of the bulk deletion.</returns>
+        public virtual async Task<Response<BulkRegistryOperationResult>> DeleteIdentitiesAsync(IEnumerable<DeviceIdentity> deviceIdentities, BulkIfMatchPrecondition precondition = BulkIfMatchPrecondition.IfMatch, CancellationToken cancellationToken = default)
+        {
+            IEnumerable<ExportImportDevice> registryOperations = deviceIdentities
+                .ToList()
+                .Select(x => new ExportImportDevice()
+                {
+                    Id = x.DeviceId,
+                    ETag = x.Etag,
+                    ImportMode = precondition == BulkIfMatchPrecondition.Unconditional ? ExportImportDeviceImportMode.Delete : ExportImportDeviceImportMode.DeleteIfMatchETag
+                });
+
+            return await _registryManagerClient.BulkDeviceCrudAsync(registryOperations, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Delete multiple devices. A maximum of 100 deletions can be done per call. For larger scale operations, consider using IoT Hub jobs (https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-identity-registry#import-and-export-device-identities).
+        /// </summary>
+        /// <param name="deviceIdentities">The devices to delete.</param>
+        /// <param name="precondition">The condition for which each operation will execute.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The result of the bulk deletion.</returns>
+        public virtual Response<BulkRegistryOperationResult> DeleteIdentities(IEnumerable<DeviceIdentity> deviceIdentities, BulkIfMatchPrecondition precondition = BulkIfMatchPrecondition.IfMatch, CancellationToken cancellationToken = default)
+        {
+            IEnumerable<ExportImportDevice> registryOperations = deviceIdentities
+                .ToList()
+                .Select(x => new ExportImportDevice()
+                {
+                    Id = x.DeviceId,
+                    ETag = x.Etag,
+                    ImportMode = precondition == BulkIfMatchPrecondition.Unconditional ? ExportImportDeviceImportMode.Delete : ExportImportDeviceImportMode.DeleteIfMatchETag
+                });
+
+            return _registryManagerClient.BulkDeviceCrud(registryOperations, cancellationToken);
+        }
+
+        /// <summary>
+        /// List a set of device twins.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>A pageable set of device twins.</returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Globalization", "CA1305:Specify IFormatProvider", Justification = "<Pending>")]
```

Comment:
I don't know what the wording for justification should be here, but probably not "pending"

Reply:
I'll just get rid of this for now. This suppression is something we can come back to later